src/Features/CSharp/Portable/UseIsNullCheck/CSharpUseIsNullCheckForCastAndEqualityOperatorDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseIsNullCheck
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpUseIsNullCheckForCastAndEqualityOperatorDiagnosticAnalyzer
+        : AbstractCodeStyleDiagnosticAnalyzer
+    {
+        public CSharpUseIsNullCheckForCastAndEqualityOperatorDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.UseIsNullCheckForCastAndEqualityOperatorDiagnosticId,
+                   CSharpFeaturesResources.Use_is_null_check,
+                   new LocalizableResourceString(nameof(FeaturesResources.Null_check_can_be_simplified), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        public override bool OpenFileOnly(Workspace workspace)
+            => false;
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxNodeAction(AnalyzeSyntax, SyntaxKind.EqualsExpression, SyntaxKind.NotEqualsExpression);
+
+        private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+
+            var semanticModel = context.SemanticModel;
+            var syntaxTree = semanticModel.SyntaxTree;
+
+            if (((CSharpParseOptions)syntaxTree.Options).LanguageVersion < LanguageVersion.CSharp7)
+            {
+                return;
+            }
+
+            var optionSet = context.Options.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).GetAwaiter().GetResult();
+            if (optionSet == null)
+            {
+                return;
+            }
+
+            var option = optionSet.GetOption(CodeStyleOptions.PreferIsNullCheckOverCastAndEqualityOperator, semanticModel.Language);
+            if (!option.Value)
+            {
+                return;
+            }
+
+            var binaryExpression = (BinaryExpressionSyntax)context.Node;
+
+            if (!IsObjectCastAndNullCheck(semanticModel, binaryExpression.Left, binaryExpression.Right) &&
+                !IsObjectCastAndNullCheck(semanticModel, binaryExpression.Right, binaryExpression.Left))
+            {
+                return;
+            }
+
+            var severity = option.Notification.Value;
+            context.ReportDiagnostic(
+                Diagnostic.Create(
+                    GetDescriptorWithSeverity(severity), binaryExpression.GetLocation()));
+        }
+
+        private static bool IsObjectCastAndNullCheck(
+            SemanticModel semanticModel, ExpressionSyntax left, ExpressionSyntax right)
```

Comment:
I get it, but I must chuckle how left and right are also passed the exact opposite. ;-)